homeworks/03-data-structure/src/BinarySearchTree.test.ts

```diff
@@ -0,0 +1,106 @@
+import { BinarySearchTree } from "./BinarySearchTree"
+import { TreeNode } from "./BinaryTree"
+
+const binaryTreeFake = {
+  value: 4,
+  left: {
+    value: 2,
+    left: {
+      value: 1,
+    },
+    right: {
+      value: 3,
+    },
+  },
+  right: {
+    value: 5,
+  },
+}
+
+const fakeCustomCompareFunction = (a: number, b: number) => {
+  if (a === b) return 0
+  if (a > b) return -1
+  return 1
+}
+
+describe("BinaryTree", () => {
+  describe("has()", () => {
+    it("returns false for emprty tree", () => {
+      const binaryTree = new BinarySearchTree()
```

Comment:
Should accept the root.

Reply:
This test case called "for emprty tree", so there is no root...

Reply:
According to the task requirements.

Reply:
I do not see such a requirement in the [task description](https://github.com/evolution-gaming/typescript-bootcamp/tree/main/homeworks/03-data-structure).